src/SWP/Bundle/ContentBundle/Service/RouteToArticleMapper.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Content Bundle.
+ *
+ * Copyright 2016 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2016 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Bundle\ContentBundle\Service;
+
+use Doctrine\ODM\PHPCR\DocumentRepository;
+use Doctrine\ORM\EntityRepository;
+use Psr\Log\LoggerInterface;
+use SWP\Bundle\ContentBundle\Model\ArticleInterface;
+use SWP\Bundle\ContentBundle\Model\RouteToArticle;
+use Symfony\Component\ExpressionLanguage\ExpressionLanguage;
+
+class RouteToArticleMapper
+{
+    /**
+     * @var EntityRepository
+     */
+    private $routeToArticleRepository;
+
+    /**
+     * @var DocumentRepository
+     */
+    private $routeRepository;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @var ExpressionLanguage
+     */
+    private $language;
+
+    public function __construct(EntityRepository $routeToArticleRepository, DocumentRepository $routeRepository, LoggerInterface $logger)
+    {
+        $this->routeToArticleRepository = $routeToArticleRepository;
+        $this->routeRepository = $routeRepository;
+        $this->logger = $logger;
+        $this->language = new ExpressionLanguage();
```

Comment:
"...if and when it needs to be." In my opinion, it will never need to be. But OK, I will create a service  with 2 lines of code.